File: [CODE]/rightoversJAVA/src/main/java/sep3/g3/rightoversjava/model/UserCreationDTO.java
package sep3.g3.rightoversjava.model;

public class UserCreationDTO
{
    private String firstName;
    private String username;
    private String password;
    private AddressCreationDTO addressCreationDTO;
    private boolean isBusiness;

    public UserCreationDTO(String firstName, String userName,
                           String password, int addressId, String streetName,
                           String streetNumber,
                           int postalCode, String cityName, double longitude, double latitude, boolean isBusiness)
    {
        this.firstName = firstName;
        this.username = userName;
        this.password = password;
        this.addressCreationDTO = new AddressCreationDTO(addressId,
                streetNumber, streetName, cityName, postalCode, longitude, latitude);
        this.isBusiness = isBusiness;
    }

    public UserCreationDTO()
    {
    }

    public String getFirstName()
    {
        return firstName;
    }

    public String getUsername()
    {
        return username;
    }

    public String getPassword()
    {
        return password;
    }

    public AddressCreationDTO getAddressCreationDTO()
    {
        return addressCreationDTO;
    }

    public boolean isBusiness() {
        return isBusiness;
    }
}

File: [CODE]/rightoversBlazorNWEB/Domain/DTOs/UserCreationDto.cs
namespace Domain.DTOs;

public class UserCreationDto
{

    public UserCreationDto(string firstName, string userName, string password, string streetName, string streetNumber,
        int postalCode, string cityName)
    {
        FirstName = firstName;
        UserName = userName;
        Password = password;
        AddressCreationDto = new AddressCreationDto(streetName, streetNumber, postalCode, cityName);
    }

    public UserCreationDto()
    {
        AddressCreationDto = new AddressCreationDto();
    }

    public string FirstName { get; set; }
    public string UserName { get; set; }
    public string Password { get; set; }
    public bool IsBusiness { get; set; }
    public AddressCreationDto AddressCreationDto { get; set; }

    public string toString()
    {
        return
            $"First name: {FirstName}, User name: {UserName}, Password: {Password}, Address: {AddressCreationDto.toString()}";
    }
}